.github/workflows/build_and_test.yml

```diff
@@ -41,30 +76,31 @@ jobs:
     name: "Build XLA CUDA plugin"
     uses: ./.github/workflows/_build_plugin.yml
     with:
-      dev-image: us-central1-docker.pkg.dev/tpu-pytorch-releases/docker/development:3.8_cuda_12.1
+      dev-image: us-central1-docker.pkg.dev/tpu-pytorch-releases/docker/development:3.10_cuda_12.1
     secrets:
       gcloud-service-key: ${{ secrets.GCLOUD_SERVICE_KEY }}
 
-  test-cpu:
-    name: "CPU tests"
-    uses: ./.github/workflows/_test.yml
-    needs: build
+  test-python-cpu:
+    name: "CPU Python tests"
+    uses: ./.github/workflows/_test_python.yml
+    needs: build-torch-xla
     with:
-      docker-image: ${{ needs.build.outputs.docker-image }}
+      dev-image: us-central1-docker.pkg.dev/tpu-pytorch-releases/docker/development:3.10_tpuvm
       timeout-minutes: 120
       collect-coverage: false
     secrets:
       gcloud-service-key: ${{ secrets.GCLOUD_SERVICE_KEY }}
 
-  test-cuda:
-    name: "GPU tests"
-    uses: ./.github/workflows/_test.yml
-    needs: build
+  test-python-cuda:
+    name: "GPU Python tests"
+    uses: ./.github/workflows/_test_python.yml
+    needs: [build-torch-xla, build-cuda-plugin]
     with:
-      docker-image: ${{ needs.build.outputs.docker-image }}
+      dev-image: us-central1-docker.pkg.dev/tpu-pytorch-releases/docker/development:3.10_cuda_12.1
```

Comment:
qq: the PR "Use TPU build for CPU and GPU Python tests" but here we are still using GPU build to run GPU python tests. Am I missing anything?